### ISSUE TITLE
Supported argument types for reverse?

### ISSUE THREAD

**Issue by avehtari** [stan-dev/docs#798]
`reverse` function doc https://mc-stan.org/docs/functions-reference/matrix_operations.html#reverse-functions says the input needs to be a `vector` or `row_vector`, but it seems to work also with `array`

**Comment by WardBrian** [comment]
This is a problem we’ve had several times based on the current organization of the docs. It does work for arrays, and this is documented (https://mc-stan.org/docs/functions-reference/array_operations.html#reversing-functions) but on a separate page, which makes discoverability hard

**Comment by WardBrian** [comment]
Yes, the current chapter structure of the functions isn't very good for google. The internal search bar does turn up all the valid options still, but it's definitely not ideal

**Comment by bob-carpenter** [comment]
Any suggestions on how we could make the structure better?  In many ways, one function per page would be nice, but then we don't see related things like all the normal distribution functions.  And there wouldn't be an easy way to scan all the matrix functions that way.

**Comment by avehtari** [comment]
Maybe just add a cross-link? That is, `matrix_operations.html#reverse-functions` could say:

> See also `reverse` functions for arrays and complex matrices.

(with hyperlinks to appropriate places, but I expect that github comment doesn't show them right)

**Comment by bob-carpenter** [comment]
> add a cross-link?

Yes, please.  That'd certainly be an improvement on what we have now.
